rest/data/project.go

```diff
@@ -96,6 +96,16 @@ func (sc *DBProjectConnector) VerifyUniqueProject(name string) error {
 	return nil
 }
 
+// GroupProjectsByRepo takes in an array of projects and groups them in a map based on the repo they are part of
+func (sc *DBProjectConnector) GroupProjectsByRepo(projects []model.ProjectRef) map[string][]model.ProjectRef {
```

Comment:
Because this function doesn't actually use the database, it shouldn't be a connector function at all. I would either do this work directly in the resolver, do the project fetching here, or add it as a regular model/project_ref.go function.